File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxP = 0;
        int min = prices[0];
        
        for (int i = 0; i<prices.length; i++){
            min = Math.min(min, prices[i]);
            maxP = Math.max(maxP, prices[i]-min);
        }
        
        return maxP;
    }
}

/*

1,2,3,4,5

min = 1


*/

// class Solution {
//     public int maxProfit(int[] prices) {
//         int maxP = 0;
//         int min = prices[0];
        
//         for (int i = 0; i<prices.length; i++){
//             if(prices[i] < min) min = prices[i];
//             else if(prices[i] - min > maxP) maxP = prices[i] - min;
//         }
        
//         return maxP;
//     }
// }

File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        # going along, find the lowest price
        
        # if we find the lowest price, then find 
        
        least = 10e4+1
        max_profit = 0
        for i in prices:
            if i < least:
                least = i
            if i-least > max_profit:
                max_profit = i-least
        return max_profit
                
                
            